### PR TITLE
Make the runner user trusted by default

### PR DESCRIPTION
**Pull request by zhaofengli**
Usually we want the runner user to be able to configure binary caches.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -159,18 +162,21 @@ runs:
...
+        nix_extra_conf=""
+        newline='
+        '
         if [ -n "${{ inputs.extra-conf }}" ]; then
-          if [ -n "${{ inputs.github-token }}" ]; then
-            export NIX_INSTALLER_EXTRA_CONF="${{ inputs.extra-conf }}access-tokens = github.com=${{ inputs.github-token }}"
-          else
-            export NIX_INSTALLER_EXTRA_CONF="${{ inputs.extra-conf }}"
-          fi
+          nix_extra_conf="${{ inputs.extra-conf }}"
```
Comment (Hoverbear) [inline action.yml:169]: Neat method. :) Do you think we can make `NEWLINE` and `NIX_EXTRA_CONF` all caps just for my own sake? :sweat_smile:

**Review by Hoverbear — APPROVED** [review]
Rest LGTM, happy to merge (preferably with my nit fixed if you don't mind!)

File: action.yml (hunk 1 above)
Comment (zhaofengli) [reply action.yml]: Sure, fixed!

File: action.yml [hunk 2]
```diff
@@ -160,18 +163,21 @@ runs:
...
-          else
-            export NIX_INSTALLER_EXTRA_CONF="${{ inputs.extra-conf }}"
-          fi
+          NIX_EXTRA_CONF="${{ inputs.extra-conf }}"
+        fi
+        if [ -n "${{ inputs.github-token }}" ]; then
+          NIX_EXTRA_CONF="${NIX_EXTRA_CONF:+$NIX_EXTRA_CONF$NEWLINE}access-tokens = github.com=${{ inputs.github-token }}"
+        fi
+        if [ "${{ inputs.trust-runner-user }}" == "true" ]; then
+          NIX_EXTRA_CONF="${NIX_EXTRA_CONF:+$NIX_EXTRA_CONF$NEWLINE}trusted-users = root $USER"
```
Comment (Hoverbear) [inline action.yml:176]: Hm, I'm not sure this is working as intended?

```bash
❯ NEWLINE='
'

❯ NIX_EXTRA_CONF="Some User Content"

❯ echo $NIX_EXTRA_CONF
Some User Content

❯ NIX_EXTRA_CONF="${NIX_EXTRA_CONF:+$NIX_EXTRA_CONF$NEWLINE}trusted-users = root $USER"

❯ echo $NIX_EXTRA_CONF
Some User Content trusted-users = root ana
```

It seems

```bash
Some User Content trusted-users = root ana
```

Should be

```bash
Some User Content
trusted-users = root ana
```

Comment (Hoverbear) [reply action.yml:176]: You know what, nevermind me:

```
❯ echo "$NIX_EXTRA_CONF"
Some User Content
 trusted-users = root ana

❯ echo $NIX_EXTRA_CONF
Some User Content trusted-users = root ana
```

**Comment by zhaofengli** [comment]
Let's anchor the regexes to line beginnings so we know for sure the newline trick is working.

**Pull request merged**